Follow up to Issue #5134

I tried recompiling my application with the very latest zig version (zig-linux-x86_64-0.6.0+9a06f966c.tar.xz) that I believe has a fix for the (now-closed) issue #5134.  However, before recompiling, I forgot to fix a variant of the incorrect type as described in #1534. I now get
```
zig build-exe --main-pkg-path .. --single-threaded --object utilities.o `pkg-config --libs gtk+-3.0` -L/usr/local/lib -L/usr/lib -lc -lsqlite3 newcash.zig
Assertion failed at /deps/zig/src/analyze.cpp:581 in get_pointer_to_type_extra2. This is a bug in the Zig compiler.
Unable to dump stack trace: debug info stripped
make: *** [makefile:4: newcash] Aborted
```
I am pretty sure the problem is this extern:
```
extern "c" fn account_tree_store_set(store:*gtk.GtkTreeStore, node_iter:*gtk.GtkTreeIter, 
    name_index:u8, name:[*:0]const u8, guid_index:u8, guid:*:0 const u8, flags_index:u8, flags:u32) void;
```
and specifically the incorrect type of the guid parameter (the compiler had suggested '*:0const u8', which produced a complaint about the 'c' immediately following the '0'; so I inserted a space to try to help matters, though the type is still incorrect; didn't help, of course, at which point I posted Issue #1534; it now produces the problem above).

There is a call of the external function in the code, like so:
```
account_tree_store_set(globals.accounts_store, &root_iter,
                       constants.account_tree_store_name_index, "Root",
                       constants.account_tree_store_guid_index, &(globals.root_account_guid)[0],
                       constants.account_tree_store_flags_index, zlib.account_flag_placeholder);
```
The type of  globals.root_account_guid is [:0]const u8.

If I correct the type of the guid parameter to '* const u8', the above error does not occur and I get a legit diagnostic about another problem in my code.

I'm posting this issue because garbage in my code provoked the compiler to issue a diagnostic claiming a compiler bug, instead of an error message pointing out my error.

--- a/src/ir.cpp
+++ b/src/ir.cpp
@@ -29066,6 +29066,12 @@ static IrInstGen *ir_analyze_instruction_ptr_type(IrAnalyze *ira, IrInstSrcPtrTy
     lazy_ptr_type->base.id = LazyValueIdPtrType;
 
     if (instruction->sentinel != nullptr) {
+        if (instruction->ptr_len != PtrLenUnknown) {
+            ir_add_error(ira, &instruction->base.base,
+                buf_sprintf("sentinels are only allowed on unknown-length pointers"));
+            return ira->codegen->invalid_inst_gen;
+        }
+
         lazy_ptr_type->sentinel = instruction->sentinel->child;
         if (ir_resolve_const(ira, lazy_ptr_type->sentinel, LazyOk) == nullptr)
             return ira->codegen->invalid_inst_gen;

--- a/test/compile_errors.zig
+++ b/test/compile_errors.zig
@@ -2,6 +2,14 @@ const tests = @import("tests.zig");
 const std = @import("std");
 
 pub fn addCases(cases: *tests.CompileErrorContext) void {
+    cases.add("invalid pointer syntax",
+        \\export fn foo() void {
+        \\    var guid: *:0 const u8 = undefined;
+        \\}
+    , &[_][]const u8{
+        "tmp.zig:2:15: error: sentinels are only allowed on unknown-length pointers",
+    });
+
     cases.add("declaration between fields",
         \\const S = struct {
         \\    const foo = 2;

